packages/amplify-gen2-codegen/src/backend/synthesizer.ts

```diff
@@ -37,6 +37,9 @@ export interface BackendRenderParameters {
     importFrom: string;
     functionNamesAndCategories: Map<string, string>;
   };
+  unsupportedCategories?: {
```

Comment:
can we have `unsupportedCategories` as a `Map` instead of having a nested property `categories` with the `Map` ?